src/decompiler-naming.h

```diff
@@ -0,0 +1,151 @@
+    if (c == '(') {
+      nesting++;
+    }
+    if (c == ')') {
+      nesting--;
+    }
+    if (nesting) {
+      continue;
+    }
+    if (!isalnum(static_cast<unsigned char>(c))) {
+      c = '_';
+    }
+    if (c == '_') {
+      if (s.empty()) {
+        continue;  // Skip leading.
+      }
+      if (s.back() == '_') {
+        continue;  // Consecutive.
+      }
+    }
+    s += c;
+    if (filter && (c == '_' || read == thing_name.size())) {
+      // We found a "word" inside a snake_case identifier.
+      auto word_end = s.size();
+      if (c == '_') {
+        word_end--;
+      }
+      assert(word_end > word_start);
+      auto word = string_view(s.c_str() + word_start, word_end - word_start);
+      if (filter->find(word) != filter->end()) {
+        s.resize(word_start);
+      }
+      word_start = s.size();
+    }
+  }
+  if (!s.empty() && s.back() == '_') {
+    s.pop_back();  // Trailing.
+  }
+  // If after all this culling, we're still gigantic (STL identifier can
+  // easily be hundreds of chars in size), just cut the identifier
+  // down, it will be disambiguated below, if needed.
+  const size_t max_identifier_length = 100;
+  if (s.size() > max_identifier_length) {
+    s.resize(max_identifier_length);
+  }
+  // Remove original binding first, such that it doesn't match with our
+  // new name.
+  bh.erase(thing_name);
+  // Find a unique name.
+  Index disambiguiator = 0;
+  auto base_len = s.size();
+  for (;;) {
+    if (bh.count(s) == 0) {
+      break;
+    }
+    disambiguiator++;
+    s.resize(base_len);
+    s += '_';
+    s += std::to_string(disambiguiator);
+  }
+  // Replace name in bindings.
+  thing_name = s;
+  bh.emplace(s, Binding(i));
+}
+
+template<typename T>
+void RenameToIdentifiers(std::vector<T*>& things, BindingHash& bh,
+                         const std::set<string_view>* filter) {
+  Index i = 0;
+  for (auto thing : things) {
+    RenameToIdentifier(thing->name, i++, bh, filter);
+  }
+}
+
+// Function names may contain arbitrary C++ syntax, so we want to
+// filter those to look like identifiers. A function name may be set
+// by a name section (applied in ReadBinaryIr, called before this function)
+// or by an export (applied by GenerateNames, called before this function),
+// to both the Func and func_bindings.
+// Those names then further perculate down the IR in ApplyNames (called after
+// this function).
+// To not have to add too many decompiler-specific code into those systems
+// (using a callback??) we instead rename everything here.
+void RenameAll(Module& module) {
+  // We also filter common C++ keywords/STL idents that make for huge
+  // identifiers.
+  // FIXME: this can obviously give bad results if the input is not C++..
+  std::set<string_view> filter = {
+    { "const" },
+    { "std" },
+    { "allocator" },
+    { "char" },
+    { "basic" },
+    { "traits" },
+    { "wchar" },
+    { "t" },
+    { "void" },
+    { "int" },
+    { "unsigned" },
+    { "2" },
```

Comment:
Why 2?
Why t? As in size_t?
These are short enough that they probably add clarity rather than get in the way

Reply:
After filtering the others, I'd end up with strings that look like `2_2_string_append_2_t_2_2` etc. For some reason internal STL namespacing used 2?

Either way, this is a very arbitrary first pass at a list by scanning thru some output.. it is to be improved later.